frontend/apps/app/features/organizations/actions/inviteMember.ts

```diff
@@ -0,0 +1,181 @@
+    .eq('email', email.toLowerCase())
+
+  return existingInvites && existingInvites.length > 0
+    ? existingInvites[0]
+    : null
+}
+
+/**
+ * Updates an existing invitation (resend)
+ */
+const updateInvite = async (
+  supabase: SupabaseClient,
+  inviteId: string,
+): Promise<void> => {
+  await supabase
+    .from('membership_invites')
+    .update({ invited_at: new Date().toISOString() })
+    .eq('id', inviteId)
+}
+
+/**
+ * Creates a new invitation
+ */
+const createInvite = async (
+  supabase: SupabaseClient,
+  organizationId: string,
+  email: string,
+  userId: string,
+): Promise<{ success: boolean; error?: string }> => {
+  const { error: insertError } = await supabase
+    .from('membership_invites')
+    .insert({
+      organization_id: organizationId,
+      email: email.toLowerCase(),
+      invite_by_user_id: userId,
+      invited_at: new Date().toISOString(),
+    })
+
+  if (insertError) {
+    return {
+      success: false,
+      error: insertError.message,
+    }
+  }
+
+  return { success: true }
+}
+
+/**
+ * Server action to invite a member to an organization
+ */
+export const inviteMember = async (formData: FormData) => {
+  // Parse and validate form data
+  const formDataObject = {
+    email: formData.get('email'),
+    organizationId: formData.get('organizationId'),
+  }
+
+  const parsedData = v.safeParse(inviteFormSchema, formDataObject)
+
+  if (!parsedData.success) {
+    return {
+      success: false,
+      error: `Invalid form data: ${parsedData.issues.map((issue) => issue.message).join(', ')}`,
+    }
+  }
+
+  const { email, organizationId } = parsedData.output
+  const supabase = await createClient()
+
+  // Get current user
+  const currentUser = await supabase.auth.getUser()
+  const userId = currentUser.data.user?.id
+
+  if (!userId) {
+    return {
+      success: false,
+      error: 'User not authenticated',
+    }
+  }
+
+  // Check if user is already a member
+  const isAlreadyMember = await checkExistingMember(
+    supabase,
+    organizationId,
+    email,
+  )
+
+  if (isAlreadyMember) {
+    return {
+      success: false,
+      error: 'This user is already a member of the organization',
+    }
+  }
+
+  // Check if invitation already exists
+  const existingInvite = await checkExistingInvite(
+    supabase,
+    organizationId,
+    email,
+  )
+
+  if (existingInvite) {
+    // Update existing invite (resend)
+    await updateInvite(supabase, existingInvite.id)
+    return { success: true }
+  }
+
+  // Create new invite
+  const createResult = await createInvite(
+    supabase,
+    organizationId,
+    email,
+    userId,
+  )
+  if (!createResult.success) {
+    return {
+      success: false,
+      error:
+        createResult.error || 'Failed to send invitation. Please try again.',
+    }
+  }
+
+  // TODO: Send email to user
+
+  revalidatePath(
+    '/(app)/app/organizations/[organizationId]/settings/members/page.tsx',
+    'page',
+  )
```

Comment:
Nits: I think something like ``revalidatePath(`/app/organizations/${organizationId}/settings/members`, 'page')`` would be preferable in terms of intent, but it might be fine to leave it as-is for now and address it when implementing the actual revalidation logic!

Reply:
You're right, this would cause all members pages of all organizations to be revalidated. I will fix this later!